docs/cicd_notes.md

```diff
@@ -1,7 +1,84 @@
+- It can be either hosted in the cloud or self-hosted.
+- It had not any recent availability problems that we could notice.
+#### Conclusion
+AppVeyor got 6 out of 6 possible points.
+
+### CircleCI
+#### Description
+A CI/CD tool that allows its free tier to use parallel builds, which is a very interesting feature that the rest of platforms except for GitHub Actions doesn't offer with their free tier.
+#### Analysis
+- **Free tier** with parallel builds.
+- They give 6000 build minutes per month for free.
+- The service was designed to be used mainly with GitHub and BitBucket, and [configuring the integration](https://circleci.com/docs/2.0/gh-bb-integration/) is fairly simple.
+- CircleCI is [configured](https://circleci.com/docs/2.0/config-intro/#section=configuration) via a YAML file that allows the user to specify all the build steps in an uncomplicated way.
+- There's no need to self-host it.
+- It had not any recent availability problems that we could notice.
+#### Conclusion
+CircleCI got 6 out of 6 points, however, since it was the service that was used the most by the rest of the students we decided not to consider it for our project.
+
+### Final choices
+After the analysis performed we decided that the best candidates to use as CI/CD platforms for our project are n**GitHub Actions** and **AppVeyor**. We discarded the other two options because, although they could be of use, SemaphoreCI suffered from critical availability problems when we first tried to use it, and CircleCI was used in almost all the rest of the projects.
+
+## Executing the unit tests with a GitHub Action
+We created a GitHub Action based on the [setup-python action](https://github.com/marketplace/actions/setup-python) that installs the project's dependencies in an ubuntu machine and then runs the unit tests for 3.10. No prior Python versions will be tested since as it was explained in the [infrastructure decisions documentation](infrastructure_decisions.md), the minimum Python version that the project supports is Python 3.9, which is the one used by the test container and will be tested using *AppVeyor*.
+Future releases of Python versions could be added to this GitHub action to ensure the project is working as expected for every supported version.
+
+## Executing the unit tests using the test container
+We configured an AppVeyor project to pass the unit tests inside the [test container](../Dockerfile). To use it, we only needed to install the GitHub App provided by the platform and give it the permissions necessary to check the repository. Once that's done the integration with GitHub is up and ready to use.
+To configure it we only needed to create the [.appveyor.yml](../.appveyor.yml) with information regarding how to configure the machine and what to execute to pass the tests. Now, the service will be automatically configured to run the unit tests every time a new commit is added to a PR.
```

Comment:
Good documentation in general. Taking advantage of the fact that you had tested some CIS, maybe you could have added screenshots or more information about the configuration of these in your project, even if in the end you decided to discard the options.

Reply:
I think that it would be interesting, and in fact I had screenshots of all the errors I got, but sadly I lost them in my local repository and I was unable to get them again :facepalm:.